ReflectometryServer/parameters.py

```diff
@@ -314,11 +426,38 @@ def __init__(self, name, component, sim=False, init=False, description=None):
         """
         if description is None:
             description = "{} component is in the beam".format(name)
-        super(InBeamParameter, self).__init__(name, sim, init, description=description)
+        super(InBeamParameter, self).__init__(name, sim, init, description, autosave)
         self._component = component
+
+        if self._autosave:
+            self._initialise_sp_from_file()
+        if self._set_point_rbv is None:
+            self._component.beam_path_set_point.add_init_listener(self._initialise_sp_from_motor)
         self._component.beam_path_rbv.add_after_beam_path_update_listener(self._trigger_rbv_listeners)
+
         self.parameter_type = BeamlineParameterType.IN_OUT
 
+    def _initialise_sp_from_file(self):
+        """
+        Read an autosaved setpoint for this parameter from the autosave file. Remains None if unsuccesful.
+        """
+        sp_init = file_io.read_autosave_param(self._name)
+        if sp_init == "True":
+            self._set_initial_sp(True)
+            self._move_component()
+        elif sp_init == "False":
+            self._set_initial_sp(False)
+            self._move_component()
+        elif sp_init is not None:
+            logger.error("Could not read autosave value for parameter {}: unexpected type.")
+
+    def _initialise_sp_from_motor(self):
+        """
+        Get the setpoint value for this parameter based on the motor setpoint position.
+        """
+        init_sp = self._component.beam_path_set_point.is_in_beam
```

Comment:
This is likely because it is being overwritten by a mode init value. I agree the current behaviour is wrong but need to think about which one should win.

Reply:
Mode init values are now not applied anymore on startup.